Automattic/wp-calypso
# WPCOM: Migrate `wpcom.undocumented()` site address change to `wpcom.req`

#### Changes proposed in this Pull Request

This PR migrates all the `wpcom.undocumented()` site address change methods to `wpcom.req.get()` and `wpcom.req.post()`. 

Part of the ongoing effort to get rid of `wpcom.undocumented()`.

#### Testing instructions

* Go to `/domains/manage/:site/change-site-address/:site` where `:site` is one of your WP.com simple sites on a `.wordpress.com` domain.
* Try inputting a valid available `.wordpress.com` domain in the form. Verify it validates properly and displays that it's available.
* Try inputting an unavailable `.wordpress.com` domain in the form. Verify it validates properly and displays that it's not available.
* Input an available domain in the form and click "Change site address". Confirm in the popup.
* Verify the 2 network requests (to `/sites/:siteId/site-address-change/nonce` and `/sites/:siteId/site-address-change` are correct, and that the address of the site changes correctly.
